CONTRIBUTING.md

```diff
@@ -1,47 +1,205 @@
+
+## Debugging
+
+### Running Cocoapods Locally
+
+Framework development is quite different from typical app development when it comes to debugging and being able to test your code. First you will want to create a new app that uses the pods you changed. For instance, if you changed something in the `Storage` category to test to see if it works, you will need to create a new app and run `pod init` and edit the `Podfile` to use the local version of Amplify that you are building on. This is done by editing the `Podfile` of the new application to take a path to the pod instead of a version number like below:
+
+```ruby
+
+target 'MySampleApp' do
+  # Comment the next line if you don't want to use dynamic frameworks
+  use_frameworks!
+  # with :path I can direct my project to use a local path for the pod
+  pod 'Amplify', :path => '~/Projects/amplify-ios'      # 
+  pod 'AmplifyPlugins/AWSS3StoragePlugin', :path => '~/Projects/amplify-ios'
+  pod 'AWSPluginsCore', :path => '~/Projects/amplify-ios'           
+  
+end
+```
+Then you want to run `pod update` at the root of your sample app to make sure it is using the local version of the pods. When you open the workspace, under the Pods project, the you'll see a section call Development Pods. This where those local pods were installed and if you want to change code here to debug, it will change in the other repo, just make sure you clean and re-build before testing your code again.
+
+## Pull Requests
+
+### Pull Request Checklist
+- Testing
+    - Unit test added or updated
+    - Integration test added or updated
+- Changelog
+    - Update Changelog with your updates in accordance with our pattern under the new version. Example below:
+
+    ```markdown
+    ## 2.xx.x
+
+    ### Bug Fixes
+
+    - **Breaking Change** List out any breaking changes here
+
+    - **Storage or name of category you changed**
+      - for example, if you made a change to Storage, add the heading like above and list the bug fix here with the issue number.
+
+    ### Misc. Updates
+
+    -  Misc updates go here. Usually model changes
+    ```
+
+### Step 1: Open Issue
+
+If there isn't one already, open an issue describing what you intend to contribute. It's useful to communicate in advance, because sometimes, someone is already working in this space, so maybe it's worth collaborating with them instead of duplicating the efforts.
+
+### Step 2: Design
+
+In some cases, it is useful and mandatory to seek for feedback on the design of your planned implementation. This is useful when you plan a breaking change or large feature, or you want advice on what would be the best path forward.
+
+The GitHub issue is sufficient for such discussions, and can be sufficient to get clarity on what you plan to do. Make sure you tag any members of the Amplify iOS team so we can help guide you: @kneekey23, @lawmicha, @wooj2, @palpatim, @royjit, @drochetti, @phani-srikar.
```

Comment:
We have the `Amplify-Native` team that can be tagged on PRs, does that work here? Might be better to phrase it that way, instead, so that we don't have to update this list if any of these people leave (or, if we want to add new people.)

Reply:
ah I didn't know we had a separate team for native, ill update this